src/scenes/AcceleratorRouter/DeliverableViewWrapper.tsx

```diff
@@ -83,21 +104,18 @@ const DeliverableViewWrapper = () => {
           onClick={() => setStatus('Approved')}
           size='medium'
         />
-        <OptionsMenu
-          onOptionItemClick={onOptionItemClick}
-          optionItems={[
-            ...(deliverable?.status === 'Not Submitted'
-              ? []
-              : [{ label: strings.NOT_SUBMITTED, value: 'not_submitted' }]),
-            ...(deliverable?.status === 'Needs Translation'
-              ? []
-              : [{ label: strings.NEEDS_TRANSLATION, value: 'needs_translation' }]),
-            ...(deliverable?.status === 'Not Needed' ? [] : [{ label: strings.NOT_NEEDED, value: 'not_needed' }]),
-          ]}
-        />
+        <OptionsMenu onOptionItemClick={onOptionItemClick} optionItems={optionItems} />
       </Box>
     );
-  }, [deliverable?.status, onOptionItemClick, setStatus, theme]);
+  }, [deliverable?.status, onOptionItemClick, optionItems, setStatus, theme]);
+
+  useEffect(() => {
+    if (requestStatus === 'success') {
+      snackbar.toastSuccess(strings.DELIVERABLE_STATUS_UPDATED);
+    } else if (requestStatus === 'error') {
+      snackbar.toastError(strings.GENERIC_ERROR);
+    }
+  }, [requestStatus, snackbar]);
```

Comment:
The useUpdateDeliverable handles the toasts. 'Approve' case has a custom message. We could move the toasts in here or keep them in the hook.

Reply:
Ah, I see, I will remove. Since this component only interacts with changing the status to "not needed" and "needs translation", the hook already does the exact same behavior.